### PR TITLE
ES6-ify Maze Classes

### PR DESCRIPTION
**Pull request by Hamms**
As preparation for the upcoming refactor/merge of Karel levels, this PR
updates all the existing Karel helper classes to use new ES6 class
declarations and syntax.

### CODE REVIEW

**Comment by Hamms** [comment]
I highly recommend viewing the diff [with whitespace ignored](https://github.com/code-dot-org/code-dot-org/pull/9775/files?w=1)

File: apps/src/maze/bee.js [hunk 1]
```diff
@@ -1,475 +1,468 @@
...
-
-var Bee = function (maze, studioApp, config) {
-  this.maze_ = maze;
-  this.studioApp_ = studioApp;
-  this.skin_ = config.skin;
-  this.defaultFlowerColor_ = (config.level.flowerType === 'redWithNectar' ?
-    'red' : 'purple');
-  if (this.defaultFlowerColor_ === 'purple' &&
-    config.level.flowerType !== 'purpleNectarHidden') {
-    throw new Error('bad flowerType for Bee: ' + config.level.flowerType);
```
Comment (joshlory) [inline apps/src/maze/bee.js:21]: Surprised that ESLint didn't catch this — is ESLint running on these files?

File: apps/src/maze/mazeMap.js [hunk 2]
```diff
@@ -117,25 +120,10 @@ MazeMap.getAllStaticGrids = function (variableGrid) {
...
 };
 
-/**
- * @return {boolean}
- */
-MazeMap.prototype.hasMultiplePossibleGrids = function () {
-  return this.staticGrids.length > 1;
-};
 
+MazeMap.deserialize = (serializedValues, cellClass) => new MazeMap(serializedValues.map(row => row.map(cellClass.deserialize)));
```
Comment (joshlory) [inline apps/src/maze/mazeMap.js]: Nit: maybe worth making this mutli-line to fit in the 80 char soft limit.  (One more below.)

**Comment by joshlory** [comment]
LGTM

File: apps/src/maze/bee.js (hunk 1 above)
Comment (Hamms) [inline apps/src/maze/bee.js:21]: Didn't catch what?

Comment (joshlory) [inline apps/src/maze/bee.js:21]: Ah I see, we don't enforce indentation.  Never mind!

Comment (Hamms) [inline apps/src/maze/bee.js:21]: Ah, yeah, and unfortunately our standard of extra-indenting continuations is also not supported by any code beautifier that I've tried.

**Pull request merged**